src/dependency-manager/manager.ts

```diff
@@ -346,16 +346,16 @@ export default abstract class DependencyManager {
     return dependency_components;
   }
 
-  validateRequiredParameters(component: ComponentConfig, parameters: Dictionary<ParameterValue>): void {
+  validateRequiredSecrets(component: ComponentConfig, secrets: Dictionary<SecretValue>): void { // TODO: 404: update
     const validation_errors = [];
-    // Check required parameters for components
-    for (const [pk, pv] of Object.entries(component.parameters)) {
-      if (pv.required !== false && parameters[pk] === undefined) {
+    // Check required parameters and secrets for components
+    for (const [key, value] of Object.entries(component.parameters).concat(Object.entries(component.secrets))) {
```

Comment:
Why does parameters still exist on ComponentConfig?

Can we remove parameters and in the transform merge the spec parameters in secrets?

Then for interpolation context set parameters = secrets.